pkg/auth/rhsso_authenticator.go

```diff
@@ -258,6 +259,34 @@ func (a *RHSSOAuthenticator) AuthURLAuth(_ string) (interface{}, error) {
 	return nil, errors.Errorf("URL Authentication not allowed for rhsso auth")
 }
 
+func (a *RHSSOAuthenticator) getInfraEnvKey(token *jwt.Token) (interface{}, error) {
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return nil, errors.Errorf("malformed token claims")
+	}
+
+	infraEnvID, ok := claims["sub"].(string)
+	if !ok {
+		return nil, errors.Errorf("token missing 'sub' claim")
+	}
+
+	infraEnv, err := common.GetInfraEnvFromDB(a.db, strfmt.UUID(infraEnvID))
+	if err != nil {
+		return nil, err
+	}
+
+	return []byte(infraEnv.ImageTokenKey), nil
+}
+
+func (a *RHSSOAuthenticator) AuthImageAuth(token string) (interface{}, error) {
+	parsedToken, err := jwt.Parse(token, a.getInfraEnvKey)
+	if err != nil {
+		return nil, err
```

Comment:
I think I should return some kind of API error here.
Right now if I give an invalid token I'm getting a 500 back. Presumably we want that to be a 401, right?

Reply:
Found this while testing my image service changes.

Reply:
Right. 500 is an absolute last resort, indicative of a failure on the service side, not the user side.